### ISSUE TITLE
APNS: Obtain individual Token Response from multiple Devices Push

### ISSUE THREAD

**Issue by epv-jbarragan** [Ph3nol/NotificationPusher#104]
First, thank you so much for sharing this library; it is a big time saver.

I am sending 2 devices in a push and I am trying to obtain the response of each Device. So far, I am only able to obtain the response of the last token sent.

Any suggestion or idea is more than welcome.

Thanks!!!

Here is the code that I am using:

```
use Sly\NotificationPusher\PushManager,
    Sly\NotificationPusher\Adapter\Apns as ApnsAdapter,
    Sly\NotificationPusher\Collection\DeviceCollection,
    Sly\NotificationPusher\Model\Device,
    Sly\NotificationPusher\Model\Message,
    Sly\NotificationPusher\Model\Push;

$pushManager = new PushManager(PushManager::ENVIRONMENT_DEV);

$apnsAdapter = new ApnsAdapter(array(
    'certificate' => base_path().'/apns_cert.pem',
    'passPhrase'  =>'Thanks for helping me!!',
));

$bad_token_device = new Device('1cf66bb3d38040d7525ac638221c1cfc6db950918d5b7995e416075828bc4850');

$good_token_device = new Device('be42a78a252ebc5c62e912643051b24e49f8ea10ab955e214076fb57f84c9bbf');

// Try good token at the beginning.
$devices = new DeviceCollection(
  array( 
    $good_token_device,
    $bad_token_device,
  ));

$message = new Message('Testing with Good Token at the beginning');

$push = new Push($apnsAdapter, $devices, $message);
$pushManager->add($push);

$collection = $pushManager->push();

foreach ($collection as $push)
{
    $adapter = $push->getAdapter();
    $response = $adapter->getResponse();
    var_dump($response->getCode() );
}
// Prints "int(8)" which means RESULT_INVALID_TOKEN

// Try bad token at the beginning.
$devices = new DeviceCollection(
  array( $bad_token_device, $good_token_device));

$pushManager = new PushManager(PushManager::ENVIRONMENT_DEV);

$message = new Message('Testing with Bad Token at the beginning');

$push = new Push($apnsAdapter, $devices, $message);
$pushManager->add($push);

$collection = $pushManager->push();

foreach ($collection as $push)
{
    $adapter = $push->getAdapter();
    $response = $adapter->getResponse();
    var_dump($response->getCode() );
}
// Prints "int(0)" which means RESULT_OK
```

**Comment by nurfgun** [comment]
same issue here

**Comment by taras-budzyn** [comment]
@epv-jbarragan @nurfgun 
Not sure if it still relevant, but maybe someone will use it
with current version there are possibility to do this (not sure about previous version of library)

So in current version (2.2.12):
- there are no $response->getCode() method
- you can do the next things

`        foreach ($collection as $push)
        {
            $adapter = $push->getAdapter();
            $response = $adapter->getResponse();

            var_dump($response->getResults());   // get array with data gsm_token to result
            var_dump($response->getSuccessCount()); // return int number of success PN
            var_dump($response->getFailureCount()); // return int number of failure PN
        }
`
getResults will provide something like this

`
array(5) {
  ["multicast_id"]=>
  int(11111111111111111)
  ["success"]=>
  int(2)
  ["failure"]=>
  int(1)
  ["canonical_ids"]=>
  int(0)
  ["results"]=>
  array(3) {
    [0]=>
    array(1) {
      ["message_id"]=>
      string(35) "0:1466088628273016%fcdb308ef3345"
    }
    [1]=>
    array(1) {
      ["error"]=>
      string(19) "InvalidRegistration"
    }
    [2]=>
    array(1) {
      ["message_id"]=>
      string(35) "0:1466088628273282%fcdb4567"
    }
  }
}
`

results fields there respond for relative gsm tokens in DeviceCollection

**Comment by theluk** [comment]
Checkout https://github.com/theluk/NotificationPusher where I implemented a ResponseCollection. Also the getResponse is not longer on the Adapter but on the Push. 

it seems like this library is not maintained any more :(

**Comment by seyfer** [comment]
This one solved by #155 and #127
@epv-jbarragan , @nurfgun if you will face such issue again - feel free to leave a comment here.
